nixos/modules/services/databases/victoriametrics.nix

```diff
@@ -1,65 +1,193 @@
+        for more information.
       '';
     };
+
     extraOptions = mkOption {
       type = types.listOf types.str;
-      default = [];
+      default = [ ];
+      example = literalExpression ''
+        [
+          "-httpAuth.username=username"
+          "-httpAuth.password=file:///abs/path/to/file"
+          "-loggerLevel=WARN"
+        ]
+      '';
       description = ''
-        Extra options to pass to VictoriaMetrics. See the README:
-        <https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/README.md>
-        or {command}`victoriametrics -help` for more
-        information.
+        Extra options to pass to VictoriaMetrics. See the docs:
+        <https://docs.victoriametrics.com/single-server-victoriametrics/#list-of-command-line-flags>
+        or {command}`victoriametrics -help` for more information.
       '';
     };
   };
+
   config = lib.mkIf cfg.enable {
     systemd.services.victoriametrics = {
       description = "VictoriaMetrics time series database";
+      wantedBy = [ "multi-user.target" ];
       after = [ "network.target" ];
       startLimitBurst = 5;
+
       serviceConfig = {
-        Restart = "on-failure";
-        RestartSec = 1;
-        StateDirectory = "victoriametrics";
+        ExecStart = startCommandLine;
         DynamicUser = true;
-        ExecStart = ''
-          ${cfg.package}/bin/victoria-metrics \
-              -storageDataPath=/var/lib/victoriametrics \
-              -httpListenAddr ${cfg.listenAddress} \
-              -retentionPeriod ${toString cfg.retentionPeriod} \
-              ${lib.escapeShellArgs cfg.extraOptions}
-        '';
-        # victoriametrics 1.59 with ~7GB of data seems to eventually panic when merging files and then
-        # begins restart-looping forever. Set LimitNOFILE= to a large number to work around this issue.
-        #
-        # panic: FATAL: unrecoverable error when merging small parts in the partition "/var/lib/victoriametrics/data/small/2021_08":
-        # cannot open source part for merging: cannot open values file in stream mode:
-        # cannot open file "/var/lib/victoriametrics/data/small/2021_08/[...]/values.bin":
-        # open /var/lib/victoriametrics/data/small/2021_08/[...]/values.bin: too many open files
+        User = "victoriametrics";
+        Group = "victoriametrics";
+        RestartSec = 1;
+        Restart = "on-failure";
+        RuntimeDirectory = "victoriametrics";
+        RuntimeDirectoryMode = "0700";
+        WorkingDirectory = workingDir;
+        StateDirectory = cfg.stateDir;
+        StateDirectoryMode = "0700";
+
+        # Increase the limit to avoid errors like 'too many open files'  when merging small parts
         LimitNOFILE = 1048576;
+
+        # Hardening
+        AmbientCapabilities = lib.mkIf (cfg.port < 1024) [ "CAP_NET_BIND_SERVICE" ];
+        CapabilityBoundingSet = if (cfg.port < 1024) then [ "CAP_NET_BIND_SERVICE" ] else [ "" ];
```

Comment:
we are missing that option now if i see it correctly.

Reply:
in case we want to keep the port option change (i dont dislike it), we might want to add a changelog for that, it did at least break my setup as i was using the listenAddress to configure the grafana datasource provisioning, leading to a broken alerting afterwards.

Reply:
I have no preference on this option either; I have revert this change & remove these two lines to avoid breaking changes.